### ISSUE TITLE
Nginx won't start

### ISSUE THREAD

**Issue by mcer12** [mariusmotea/diyHue#302]
Hi, after fresh diyhue install I'm unable to discover the bridge whatsoever. The nginx server won't start, here's the error

Jul 24 23:08:53 orangepione systemd[1]: Starting A high performance web server and a reverse proxy server...
Jul 24 23:08:53 orangepione nginx[4307]: nginx: [emerg] BIO_new_file("/opt/hue-emulator/public.crt") failed (SSL: error:02001002:system library:fopen:No 
Jul 24 23:08:53 orangepione nginx[4307]: nginx: configuration file /etc/nginx/nginx.conf test failed
Jul 24 23:08:53 orangepione systemd[1]: nginx.service: Control process exited, code=exited status=1
Jul 24 23:08:53 orangepione systemd[1]: Failed to start A high performance web server and a reverse proxy server.
Jul 24 23:08:53 orangepione systemd[1]: nginx.service: Unit entered failed state.
Jul 24 23:08:53 orangepione systemd[1]: nginx.service: Failed with result 'exit-code'.

Any thoughts of what this could be?

**Comment by danieledwardgeorgehitchcock** [comment]
Possibly the same as my NGINX error - port 80 is already in use elsewhere on your system

**Comment by mcer12** [comment]
I'm not running any other server on that port. I have webcam running on 8080 and 8081 but I already tried to stop the service.

**Comment by mcer12** [comment]
Returned only 8080 and 8081... there must be some different problem

**Comment by mariusmotea** [comment]
sudo nginx -t to see the error? Yesterday i have some downtime and certificate generation service did not work. The updated easy-install script mult detect this and ask to retry later.

**Comment by mariusmotea** [comment]
Execute again installation script to perform upgrade and dix certificate issue

**Comment by mcer12** [comment]
I should add that I already tried to run installation several times.
sudo nginx -t returns this:

> nginx: [emerg] BIO_new_file("/opt/hue-emulator/public.crt") failed (SSL: error:0        2001002:system library:fopen:No such file or directory:fopen('/opt/hue-emulator/        public.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
> nginx: configuration file /etc/nginx/nginx.conf test failed
>

**Comment by mariusmotea** [comment]
in configuration is:
```
                ssl_certificate /opt/hue-emulator/public.crt;
                ssl_certificate_key /opt/hue-emulator/private.key;
```
Do you have the files /opt/hue-emulator/public.crt and /opt/hue-emulator/private.key ?Did you execute again the easy_install script in order to see if this will be solved?

**Comment by mcer12** [comment]
yes I tried that. The files are there, I also tried to change their permissions but it didnt help.

**Comment by mariusmotea** [comment]
Here is something strange:
`No such file or directory:fopen('/opt/hue-emulator/ public.crt','r')`
it seams there is one space character in front of public.crt.

**Comment by mcer12** [comment]
I'm pretty sure it was just a line break when I copied it from terminal. I double checked it and there is no space. The file is empty though is it supposed to? Im not very familiar with how nginx works so I can't tell. Maybe it creates the file just fine but fails to write/generate the certificate?

**Comment by mariusmotea** [comment]
Delete both files if are empty, run easy install again to download fresh certificates

**Comment by mcer12** [comment]
Sorry for the late reply. It didn't help...

**Comment by mariusmotea** [comment]
Now i'm running a test version that don't require nginx. I will made this public in one day if i don't found any problems.

**Comment by mariusmotea** [comment]
In develop branch is not the version that don't require nginx. This will not work directly because it work with pem certificate file and you must have separate public.crt and private.key from previews deployment. The structure is identical, only different is that the pem file has the content of both .crt and .key files (i don't believe the order is important). Also ensure you totally remove nginx, on my rasperrypi this start on boot after i remove nginx package. I manually delete the systemd file in order to get rid of nginx.
